### ISSUE TITLE
Malfromed Contract. Found Requires after assignment

### ISSUE THREAD

**Issue by voroninp** [Microsoft/CodeContracts#29]
After abbreviators [started to work](https://github.com/Microsoft/CodeContracts/issues/23), we get the error 
>Malfromed Contract. Found Requires after assignment.

for the following method:

```C#
[ContractAbbreviator]
[SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "value")]
[SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "userMessage")]
public static void NotDefault<T>(T value, string userMessage)
{
	Contract.Requires<ArgumentException>(!EqualityComparer<T>.Default.Equals(value, default(T)), userMessage);
}
```

If method's body is inlined as a usual CodeContract, or abbreviator is used within the same assembly no errors appear.

**Update**
This error occurs only for enum types.

**Comment by kwhitefoot** [comment]
Happens in VB with VS 2013 update 4 and contracts.1.7.11202.10 if I declare a field with an initializer.
for instance:

class x
  private y as integer = 42

public function z(a as integer) as integer
  contract.requires( 0 < a)
....

Compiling with contracts enabled gives a malformed contract error point at the requires in public function z.

**Comment by hubuk** [comment]
default(T) (initobj) expression is causing the problem as it is interpreted as assignment:
[CodeContracts/System.Compiler/Reader.cs:4619](https://github.com/Microsoft/CodeContracts/blob/master/System.Compiler/Reader.cs#L4619-L4623)
Problem is reproducible even if the abbreviator is in the same assembly.

**Comment by hubuk** [comment]
Should be fixed in  PR #54 (Fixed usage of deafult(T) in contract abbreviators.)

**Comment by hubuk** [comment]
This issue may be closed as already fixed.
